### DISCUSSION TITLE
Deleting missed jobs

### DISCUSSION THREAD

**Discussion by belei-bohdan** [quartznet/quartznet#1439] [Q&A] [answered]
Hello! We are using quartznet v3. After deleting the implementation of the job (class that implements IJob interface) old job_details remain in database. Is it possible to configure the cleaning of old job_details after removing the implementation?

**Comment by lahma** [comment, marked as answer]
You should probably also inform the scheduler that you deleted the type from .NET system. Ideally you would do that before deleting the actual type from you code base, using scheduler's methods. But if you have already deleted type and there's pending data (job is durable or triggers referring to it), you need to delete it manually - make sure you delete both job detail information and triggers referring to it.

So general advice is to mark your old implementation as `[Obsolete]` for a while and make sure that your deployment process deletes such data gracefully.

**Reply by belei-bohdan** [reply]
Thanks for the answer.
